backend/src/test/java/click/snekhome/backend/controller/IntegrationTest.java

```diff
@@ -792,8 +798,70 @@ void expectUnauthorizedWhenLoginWithBadCredentials() throws Exception {
     }
 
     @Test
+    @DirtiesContext
     void expectUnauthorizedWhenLoginWithoutUser() throws Exception {
         mockMvc.perform(MockMvcRequestBuilders.post("/api/user/login").with(csrf()))
                 .andExpect(MockMvcResultMatchers.status().isUnauthorized()).andReturn();
     }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser(username = "test")
+    void expectNodesOfOwnerWhenGettingNodesByOwnerId() throws Exception {
+        Node node1 = new Node("abc", "123", "Home", 1, 100, new Coordinates(0, 0, 0), 0, 0);
+        Node node2 = new Node("def", "123", "Office", 2, 100, new Coordinates(0, 0, 0), 0, 0);
+        nodeRepo.save(node1);
+        nodeRepo.save(node2);
+        String expected = """
+                [
+                    {
+                        "id":"abc",
+                        "name":"Home",
+                        "level":1,
+                        "health":100,
+                        "ownerId": "123",
+                        "coordinates": {
+                            "latitude": 0,
+                            "longitude": 0
+                            },
+                        "lastUpdate":0,
+                        "lastAttack":0
+                        },
+                    {
+                        "id":"def",
+                        "name":"Office",
+                        "level":2,
+                        "health":100,
+                        "ownerId": "123",
+                        "coordinates": {
+                            "latitude": 0,
+                            "longitude": 0
+                            },
+                        "lastUpdate":0,
+                        "lastAttack":0
+                        }
+                ]
+                """;
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/nodes/123"))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().json(expected));
+    }
+
+    @Test
+    @DirtiesContext
+    void expectPlayerWithIncreasedCreditsWhenCreditsAreGenerated() {
+        Node node1 = new Node("abc", "abc", "Home", 1, 100, new Coordinates(0, 0, 0), 0, 0);
+        nodeRepo.save(node1);
+        Player player = new Player("abc", "123", "playerunknown", new Coordinates(0, 0, 0), 1, 0, 100, 100, 100, 5, 15, 0);
+        playerRepo.save(player);
+        nodeService.generateCredits();
+        Optional<Node> ownedNode = nodeRepo.findById("abc");
+        Optional<Player> playerById = playerRepo.findById("abc");
+        Optional<Player> playerByName = playerRepo.findPlayerByName("playerunknown");
+        assertTrue(ownedNode.isPresent());
+        assertTrue(playerById.isPresent());
+        assertTrue(playerByName.isPresent());
+        Player updatedPlayer = playerByName.get();
+        assertEquals(100, updatedPlayer.credits());
+    }
```

Comment:
I am not sure, but these doesn't look like an Integration test, this looks more like a service test

Reply:
I'm not mocking anything and all the necessary classes are in this Integration test so I put it here😅